tiledb/sm/c_api/tiledb.h

```diff
@@ -4755,6 +4769,31 @@ TILEDB_EXPORT int32_t tiledb_array_reopen_at(
 TILEDB_EXPORT int32_t tiledb_array_get_timestamp(
     tiledb_ctx_t* ctx, tiledb_array_t* array, uint64_t* timestamp);
 
+/**
+ * Sets the array config.
+ *
+ * **Example:**
+ *
+ * @code{.c}
+ * tiledb_array_t* array;
+ * tiledb_array_alloc(ctx, "s3://tiledb_bucket/my_array", &array);
+ * tiledb_array_open(ctx, array, TILEDB_READ);
+ * // Set the config for the given array.
+ * tiledb_config_t* config;
+ * tiledb_array_set_config(ctx, array, config);
+ * @endcode
+ *
+ * @param ctx The TileDB context.
+ * @param array The array to set the config for.
+ * @param config The config to be set.
+ * @return `TILEDB_OK` for success and `TILEDB_ERR` for error.
+ *
+ * @note The array does not need to be opened via `tiledb_array_open_at` to use
+ *      this function.
+ */
+TILEDB_EXPORT int32_t tiledb_array_set_config(
```

Comment:
We need to add a note to this function as well as `tiledb_array_open_*` that the config should be set before ioening the array, if the user wants the config parameters to take effect. In addition, it may be a good idea to throw an error if this function is called after the array is opened.